File: sdks/android/src/main/java/org/nanoai/NanoRuntime.java
package org.nanoai;

public class NanoRuntime {
    private long nativeHandle;

    static {
        System.loadLibrary("nanoai");
    }

    public NanoRuntime() {
        this.nativeHandle = nativeInit();
    }

    public void close() {
        if (nativeHandle != 0) {
            nativeDestroy(nativeHandle);
            nativeHandle = 0;
        }
    }

    @Override
    protected void finalize() throws Throwable {
        close();
        super.finalize();
    }

    public boolean loadModel(String modelPath) {
        if (nativeHandle == 0) return false;
        return nativeLoadModel(nativeHandle, modelPath);
    }

    public String generate(String prompt) {
        if (nativeHandle == 0) return "Runtime not initialized";
        return nativeGenerate(nativeHandle, prompt);
    }

    // Native methods
    private native long nativeInit();
    private native void nativeDestroy(long handle);
    private native boolean nativeLoadModel(long handle, String modelPath);
    private native String nativeGenerate(long handle, String prompt);
}

File: sdk/kotlin/src/main/kotlin/org/nanoai/NanoRuntime.kt
package org.nanoai

class NanoRuntime : AutoCloseable {
    private var handle: Long = 0

    init {
        System.loadLibrary("nanoai")
        handle = nativeInit()
    }

    override fun close() {
        if (handle != 0L) {
            nativeDestroy(handle)
            handle = 0
        }
    }

    fun loadModel(path: String, id: String): Boolean = nativeLoadModel(handle, path, id)
    fun generate(prompt: String, id: String, priority: Int): String = nativeGenerate(handle, prompt, id, priority)

    private external fun nativeInit(): Long
    private external fun nativeDestroy(handle: Long)
    private external fun nativeLoadModel(handle: Long, path: String, id: String): Boolean
    private external fun nativeGenerate(handle: Long, prompt: String, id: String, priority: Int): String
}
